src/Collection.php

```diff
@@ -520,9 +520,10 @@ public function find($filter = [], array $options = [])
             $options['typeMap'] = $this->typeMap;
         }
 
-        $operation = new Find($this->databaseName, $this->collectionName, $filter, $options);
+        //$operation = new Find($this->databaseName, $this->collectionName, $filter, $options);
+        //return $operation->execute($server);
 
-        return $operation->execute($server);
+        return new \MongoDB\Cursor($this, $filter, $options);
```

Comment:
Also, i didnt mean a general purpose compatbility actually. I just wanted to make some simple additions for ease of use, also compatible with legacy version.